### DISCUSSION TITLE
LineIntersector segment stack changes??

### DISCUSSION THREAD

**Discussion by rolandhill** [vsg-dev/VulkanSceneGraph#1101] [Q&A] [answered]
Hi Robert,

I just updated all the VSG libraries today for the first time in a couple of months and found that my line intersectors are not working any more. I use MatrixTransforms to offset groups of vertices (OctTree), so the vertex and intersector data must be consistently either in local or in world coordinates. It used to all work in local coordinates, but now none of my intersections work and it looks like the vsg::LineSegmentIntersector::_lineSegmentStack is storing world coordinates. Is this the case? If so, is this a bug or an intended change? 

Apologies if I've missed a post about it.

Regards,
Roland

**Comment by robertosfield** [comment]
Sorry about this. Another user reported an issue like this yesterday. Looks like a regression around the time I merged the Animation work. It's the first thing I'll tackle today.

**Comment by robertosfield** [comment]
I have used git bisect this morning to figure out when the regression happened and have pin pointed it to:

b98eec4aa8aef255dbe920e27ead96c496065b59

**Comment by robertosfield** [comment]
I have created Issue #1102 for this regression.  I will examine the problem commit so see why it causes the bug, I should be able to figure out a fix this morning.

**Comment by robertosfield** [comment, marked as answer]
I have tracked down the regression to the previous use of default copy constructor, that's not possible when using CopyOp so I have implemented the copy constructor manually:

1380b6ab6cf079d2ae3ff1e2185cd07e258d7ac9

**Comment by rolandhill** [comment]
Fixed by Robert in the issue mentioned above. I'll close this now.
